contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -437,10 +436,22 @@ class Resource extends mix(APIResource, IndexedDBResource) {
     API_RESOURCES[urlName] = this;
     // Overwrite the false default for IndexedDBResource
     this.syncable = syncable;
+    // A map of stringified request params to a last fetched time and a promise
+    this._requests = {};
   }
 
   fetchCollection(params) {
-    return client.get(this.collectionUrl(), { params }).then(response => {
+    const now = Date.now();
+    const queryString = paramsSerializer(params);
+    if (
+      this._requests[queryString] &&
+      this._requests[queryString][LAST_FETCHED] &&
+      this._requests[queryString][LAST_FETCHED] + REFRESH_INTERVAL * 1000 > now &&
+      this._requests[queryString].promise
+    ) {
+      return this._requests[queryString].promise;
```

Comment:
Perhaps assign `this._requests[queryString]` to a variable to save some bytes in minification.

Reply:
Yeah, wfm.